File: 02-Arrarys/Pset/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Compute the Coleman-Liau index
    double L=1.0*letters/words*100;
    double S=1.0*sentences/words*100;
    double index=0.0588*L-0.296*S-15.8;

    int level=round(index);

    // Print the grade level
    if(level>=16)
        printf("Grade 16+\n");
    else if(level<1)
        printf("Before Grade 1\n");
    else
        printf("Grade %d\n",level);

    return 0;
}

int count_letters(string text)
{
    // Return the number of letters in text
    int LENGTH=strlen(text);
    int count=0;

    for(int i=0;i<LENGTH;i++)
        if(isalnum(text[i]))
            count++;

    return count;
}

int count_words(string text)
{
    // Return the number of words in text
    int LENGTH=strlen(text);
    int count=0;

    for(int i=0;i<LENGTH;i++)
        if(isblank(text[i]))
            count++;

    return count+1;
}

int count_sentences(string text)
{
    // Return the number of sentences in text
    int LENGTH=strlen(text);
    int count=0;

    for(int i=0;i<LENGTH;i++)
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
            count++;

    return count;
}

File: 06 - Python/Pset/sentimental-readability/readability.py
def count_letters(text):
    return sum(1 for c in text if c.isalpha())


def count_words(text):
    return len(text.split())


def count_sentences(text):
    return sum(1 for c in text if c in '.!?')


def coleman_liau_index(letters, words, sentences):
    L = letters / words * 100
    S = sentences / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    return round(index)


def main():
    text = input("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    grade = coleman_liau_index(letters, words, sentences)

    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


if __name__ == "__main__":
    main()
